content/blog/2020-03-01-bookdown-learnings/index.Rmd

```diff
@@ -0,0 +1,257 @@
+- I used happygitwithR "[Connect RStudio to Git and GitHub](https://happygitwithr.com/rstudio-git-github.html)" to do exactly that. 
+Years ago I had already used [happygitwithR](https://happygitwithr.com/) to set up my git/GitHub from square-one and my https credentials.
+- Maëlle and I both worked on feature branches of master. 
+On any given day we each might work on mulitple chapters of the book. Because each book chapter is generated from one Rmd file (more on that in tip #4), and you don't want one pull request to be _too_ complex, I would create a separate feature branch for any chapter or file I worked on. I would periodically push my work to GitHub in a pull request (more in tip #3) with "_Draft_" status so Maëlle could see I was not yet seeking her review. At the end of a day's commits, I would update the pull request status to "_Ready for review_" and assign Maëlle as reviewer. She would review, edit, and merge my pull requests and open new branches of her own for me to review.
+
+#### 3. How to usethis
+
+At the start of the project, Maëlle asked me if I used the usethis package. 
+I did not. 
+But I bought into it because "[usethis](https://usethis.r-lib.org/) is a workflow package: it automates repetitive tasks that arise during project setup and development, both for R packages and non-package projects."
+The trickiest part, for me, was commiting to setting up and forcing myself to use usethis pull request helpers despite not really understanding how it would help me. 
+
+From previous work I had already followed happygitwithr to set up my [GitHub authentication via https](https://happygitwithr.com/credential-caching.html) and a [GitHub Personal Access Token (PAT)](https://happygitwithr.com/github-pat.html) in my `.Renviron`.
+I followed usethis [setup instructions](https://usethis.r-lib.org/articles/articles/usethis-setup.html) to edit my `.Rprofile` so that its functions would be available without explicitly calling the package.
+I also followed the rest of the setup instructions somewhat religiously (e.g. "Prepare your system to build packages from source"), because I wasn't confident enough to make a judgement on what I would and would not need.
+
+Practically, I kept the usethis [pull request helpers](https://usethis.r-lib.org/articles/articles/pr-functions.html) article open in my browser and forced myself to use them.
+At the start of the day, I would open our bookdown RStudio project by clicking my local `ropensci-blog-guidance.Rproj` file, pull to update my local master (because Maëlle would have reviewed, edited and merged my pull requests while I slept), create a new branch with `pr_init(branch = "branchname")`, make edits, and `pr_push()` to push my local changes to GitHub master.
+usethis automatically opened a browser to the GitHub web interface prompting me to open a pull request. 
```

Comment:
In the end, was it worth it? This tip lacks a conclusion. :-)

Reply:
You caught me. Lack of conclusion was glaring to me because I think it was worth it, but it's tied up in my new use of RStudio so hard to separate the value of usethis alone.

Reply:
Maybe you could still say whether the PR workflow feels as good as what you do in the terminal?